### ISSUE TITLE
Toybox Seems to Block Level-up UI

### ISSUE THREAD

**Issue by Bahbus** [cabarius/ToyBox#632]
**Describe the bug**
After most recent Toybox and WotR updates, receiving the next level the MC was able to pull up the UI no problem, but for companions the UI freezes. A UI Reset can be given to successfully unstick it, however, re-attempting to level-up a companion relocks it. This was the level-up to 7.

**Version Info:**
 - ToyBox Version: 1.4.14
 - OS: Windows 11
 - Game: 1.2.0h

**Additional context**
Disabling Toybox at main menu and reloading the save allowed the UI to pop back up. Will continue to test if it keeps happening beyond this level. Will add save file if this is more unique of an issue.

**Comment by Zanazar** [comment]
I have a similar issue, can't level up any character, screen just gets a little darker and nothing else happens. Resetting the UI gives me back control but I can't open the level up screen at all. The mythic level up screen works normally. I am using gestalt characters so that might be why it doesn't work at all but I tried it on a non gestalt character and it was the same.

**Comment by thehambeard** [comment]
It seems to me that when the character is in "auto level up choices" mode then the UI hangs.

**Comment by Zanazar** [comment]
> It seems to me that when the character is in "auto level up choices" mode then the UI hangs. Once you have made choices for the character during a level up then the level UI will work normally.

Playing on core difficulty, so no auto level up. I will try a new character later to test if the issue persists.

**Comment by thehambeard** [comment]
> Playing on core difficulty, so no auto level up. I will try a new character later to test if the issue persists.

If looking to recreate the behavior trying to level up a character with auto level up will cause the game to hang though.  It appears that the UI is never bound.  Turning on the UI manually with Unity Explorer does not fix the issue.

**Comment by thehambeard** [comment]
```
[HarmonyPatch(typeof(IgnorePrerequisites), nameof(IgnorePrerequisites.Ignore), MethodType.Getter)]
        private static class IgnorePrerequisites_Ignore_Patch {
            private static void Postfix(ref bool __result) {
                var state = Game.Instance.LevelUpController.State;

                if (!state.IsClassSelected) {
                    if (settings.toggleIgnoreClassRestrictions) {
                        __result = true;
                    }
                }
                else {
                    if (settings.toggleIgnoreFeatRestrictions) {
                        __result = true;
                    }
                }
            }
        }
```
`var state = Game.Instance.LevelUpController.State;` is `null` meaning the LevelUpController is not getting started properly.

**Comment by thehambeard** [comment]
After cloning the current code base and compiling the level up works as intended.  Perhaps the released version was not compiled against the newest assemblies.

**Comment by thehambeard** [comment]
Can you test this? This is with the section commented out.  This will disable ignore class restrictions and ignore feat restrictions as well.
[ToyBox1.4.14a-betafix](https://github.com/thehambeard/ToyBox/releases/tag/1.4.14a-betafix)

**Comment by Zanazar** [comment]
Hey, as I said it magically fixed itself after reinstalling everything, but today I tried playing and the game doesn' even start -.- even without mods... so unfortunately I can't test it right now :/

**Comment by Phoenix4O4** [comment]
> Can you test this? This should fix it. https://github.com/thehambeard/ToyBox/releases/tag/1.4.14b-betafix

Yes, your release works perfectly, the lvlup interface opens correctly

**Comment by kenken244** [comment]
I'm still getting this issue on 1.4.15
